sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/FunctionToolCallingSample.java

```diff
@@ -0,0 +1,248 @@
+import com.azure.ai.openai.models.FunctionProperties;
+import com.azure.core.credential.AzureKeyCredential;
+import com.azure.core.util.BinaryData;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Sample demonstrates how to do a single and parallel function calling via 'tools' calling.
+ */
+public class FunctionToolCallingSample {
+
+    /**
+     * Runs the sample algorithm and demonstrates how to get chat completions using function tool calling.
+     *
+     * @param args Unused. Arguments to the program.
+     */
+    public static void main(String[] args) {
+        String endpoint = "{azure-open-ai-endpoint}";
+        String azureOpenaiKey = "{azure-open-ai-key}";
+        // Newer models like gpt-4-1106-preview or gpt-3.5-turbo-1106 can call multiple functions in one turn.
+        String deploymentOrModelId = "{azure-open-ai-deployment-model-id}";
+
+        // Function call is supported for model versions with suffix `-0613` and later.
+        OpenAIClient client = new OpenAIClientBuilder()
+                .endpoint(endpoint)
+                .credential(new AzureKeyCredential(azureOpenaiKey))
+                .buildClient();
+
+        singleFunctionCallingSample(client, deploymentOrModelId);
+        parallelFunctionCallingSample(client, deploymentOrModelId);
+
+    }
+
+    private static void singleFunctionCallingSample(OpenAIClient client, String deploymentOrModelId) {
+        List<ChatRequestMessage> chatRequestMessages = new ArrayList<>();
+        chatRequestMessages.add(new ChatRequestSystemMessage("Don't make assumptions about what values to plug into functions. Ask for clarification if a user request is ambiguous."));
+        chatRequestMessages.add(new ChatRequestUserMessage("Give me a weather report for Toronto, Canada."));
+
+        // A list of tools to use (currently only has one tool type, function tool).
+        ChatCompletionsOptions chatCompletionsOptions = new ChatCompletionsOptions(chatRequestMessages)
+                .setTools(
+                        Arrays.asList(
+                                getCurrentWeatherToolFunction,
+                                getNDayWeatherForecastToolFunction
+                        )
+                );
+
+        // We can force the model to use a specific function, for example get_n_day_weather_forecast by using the
+        // function_call argument. By doing so, we force the model to make assumptions about how to use it.
+        chatCompletionsOptions.setToolChoice(BinaryData.fromObject(getNDayWeatherForecastToolFunction));
+
+        // We can also force the model to not use a function at all. By doing so we prevent it from producing a proper function calling.
+        // chatCompletionsOptions.setToolChoice(BinaryData.fromString("none"));
```

Comment:
This comment is a bit confusing, as it doesn't explain the code around it, it rather comments on a different aspect of the feature. If you want, you could add a separate sample with `ToolChoice::None`, otherwise I would remove it.